recipes/armadillo/all/conanfile.py

```diff
@@ -0,0 +1,414 @@
+import os
+
+
+class ArmadilloConan(ConanFile):
+    name = "armadillo"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "http://arma.sourceforge.net"
+    description = "Armadillo is a high quality C++ library for linear algebra and scientific computing, aiming towards a good balance between speed and ease of use."
+    topics = (
+        "linear algebra",
+        "scientific computing",
+        "matrix",
+        "vector",
+        "math",
+        "blas",
+        "lapack",
+        "mkl",
+        "hdf5",
+    )
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "use_blas": [
+            False,
+            "openblas",
+            "system_openblas",
+            "system_blas",
+            "system_mkl",
+            "system_flexiblas",
+            "framework_accelerate",
+        ],
+        "use_lapack": [
+            False,
+            "openblas",
+            "system_openblas",
+            "system_lapack",
+            "system_mkl",
+            "system_atlas",
+            "framework_accelerate",
+        ],
+        "use_hdf5": [False, "hdf5", "system_hdf5"],
+        "use_superlu": [False, "system_superlu"],
+        "use_extern_rng": [True, False],
+        "use_arpack": [False, "system_arpack"],
+        "use_wrapper": [True, False],
+        "mkl_library_path": "ANY",
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "use_blas": "openblas",
+        "use_lapack": False,
+        "use_hdf5": "hdf5",
+        "use_superlu": False,
+        "use_extern_rng": False,
+        "use_arpack": False,
+        "use_wrapper": False,
+        "mkl_library_path": "",
+    }
+    # Values that must be set for multiple options to be valid
+    co_dependencies = {
+        "system_mkl": [
+            "use_blas",
+            "use_lapack",
+        ],
+        "framework_accelerate": [
+            "use_blas",
+            "use_lapack",
+        ],
+    }
+    exports_sources = "CMakeLists.txt"
+    generators = (
+        "cmake",
+        "cmake_find_package",
+    )
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        if self.settings.os == "Macos":
+            # Macos will default to Accelerate framework
+            self.options.use_blas = "framework_accelerate"
+            self.options.use_lapack = "framework_accelerate"
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+        # Use default MKL path or throw error if no default path exists
+        if self.options.use_blas == "system_mkl" and not self.options.mkl_library_path:
```

Comment:
That's why system libs should not be used, it's fragile and probably we will see people reporting problems with their environment.

Reply:
I think in this case, this is an important option to include. CCI will never package MKL because it's closed source - see #6801. However, it's a _very efficient_ implementation of the BLAS/LAPACK interfaces, so a lot of people prefer this as a backend to some of the open source alternatives. It does reduce reproducibility, but I think it adds a lot of value to consumers to have the option available.